client/rpc-api/src/state/mod.rs

```diff
@@ -31,7 +31,10 @@ pub use self::gen_client::Client as StateClient;
 
 /// Substrate state API
 #[rpc]
-pub trait StateApi<Hash> {
+pub trait StateApi<Hash>
+	where
```

Comment:
Maybe drop `where` and add it directly to the generic parameter declaration?
```
pub trait StateApi<Hash: Clone>
```

I also think we should use `Clone` not `Copy`, as it's less restrictive.

Reply:
jsonrpc gives this weird errror

```
error[E0658]: associated type bounds are unstable
  --> client/rpc-api/src/state/mod.rs:34:20
   |
34 | pub trait StateApi<Hash: Clone> {
   |                    ^^^^^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/52662

error[E0107]: wrong number of type arguments: expected 1, found 0
  --> client/rpc-api/src/state/mod.rs:33:1
   |
33 | #[rpc]
   | ^^^^^^ expected 1 type argument

error[E0229]: associated type bindings are not allowed here
  --> client/rpc-api/src/state/mod.rs:34:20
   |
34 | pub trait StateApi<Hash: Clone> {
   |                    ^^^^^^^^^^^ associated type not allowed here

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0107, E0229, E0658.
For more information about an error, try `rustc --explain E0107`.
error: could not compile `sc-rpc-api`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```